backend/services/implementations/courseService.ts

```diff
@@ -0,0 +1,121 @@
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import {
+  CourseRequestDTO,
+  CourseResponseDTO,
+  ICourseService,
+} from "../interfaces/ICourseService";
+import MgCourse, { Course } from "../../models/course.model";
+
+const Logger = logger(__filename);
+
+class CourseService implements ICourseService {
+  async getCourse(id: string): Promise<CourseResponseDTO> {
+    let course: Course | null;
+    try {
+      course = await MgCourse.findById(id);
+      if (!course) {
+        throw new Error(`Course id ${id} not found`);
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get course. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
+      id: course.id,
+      title: course.title,
+      description: course.description,
+      image: course.image,
+      previewImage: course.previewImage,
+      lessons: course.lessons,
+    };
+  }
+
+  async getCourses(): Promise<CourseResponseDTO[]> {
+    try {
```

Comment:
Linter is going to yell at you for not using "this" in a class method. Technically, these methods should be static methods, but really doesn't matter. Just add `/* eslint-disable class-methods-use-this */` on the line above the getCourses and createCourse functions to stop the linter yelling at you, or use static methods, up to you.

Reply:
Probably should keep it consistent in the repo. Thoughts on changing everything to use `this` vs just bypassing the linter?

Reply:
I don't really see how you'd use `this` inside the method? There's no state being kept in the service itself. The pattern in the repo seems to be to bypass the linter in this case, look at the authService.ts file for reference.

Reply:
I think changing it to `this` might be a bit cleaner instead of disabling the linter, but I can do that when I set up the linter this week. Doesn't matter too much for the PR